src/component/reducer/actions/DomainActions.ts

```diff
@@ -73,18 +74,23 @@ function getDomain(drfat: Draft<State>) {
     xDomains,
   };
 }
-function get2DDomain(state) {
+function get2DDomain(state: State) {
   let xArray = [];
   let yArray = [];
   let yDomains = {};
   let xDomains = {};
 
-  const { activeTab, tabActiveSpectrum, data } = state;
+  const {
+    view: {
+      spectra: { activeSpectra, activeTab },
+    },
+    data,
+  } = state;
 
   const nucleus = activeTab.split(',');
 
   try {
-    xArray = data.reduce((acc, datum: Datum1D | Datum2D) => {
+    xArray = data.reduce((acc: any, datum: Datum1D | Datum2D) => {
```

Comment:
Can you fix this use of `any`? `reduce` is a generic that takes the type of the accumulated value:
`data.reduce<Array<[number, number]>>(...)`